core/array/shared/delete_if.rb

```diff
@@ -0,0 +1,27 @@
+describe :delete_if, shared: true do
+  before :each do
+    @object = [1,2,3]
+  end
+
+  ruby_version_is "2.3" do
+    it "updates the receiver after all blocks" do
+      @object.send(@method) do |i|
```

Comment:
Small nitpick: could you update the block variable to be `e`?
`i` sounds like index but here it's an element.